.github/workflows/gosec.yml

```diff
@@ -0,0 +1,38 @@
+name: Run Gosec
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
+  push:
+    branches:
+      - main
```

Comment:
Is it possible to add all branches under `feature`? (i.e., `feature/*` or something similar)

Reply:
Also, any reason for not running this on throttle branches?

Reply:
@thanethomson made suggestions for the branches so I added them. However, y'all free to add as many branches after we've merged this PR :-)